src/wallet/wallet.cpp

```diff
@@ -3437,6 +3437,45 @@ void CWallet::ListLockedCoins(std::vector<COutPoint>& vOutpts)
     }
 }
 
+
+// Note Locking Operations
+
+void CWallet::LockNote(JSOutPoint& output)
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    setLockedNotes.insert(output);
+}
+
+void CWallet::UnlockNote(JSOutPoint& output)
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    setLockedNotes.erase(output);
+}
+
+void CWallet::UnlockAllNotes()
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    setLockedNotes.clear();
+}
+
+bool CWallet::IsLockedNote(uint256 hash, size_t js, uint8_t n) const
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    JSOutPoint outpt(hash, js, n);
```

Comment:
Pass the `JSOutPoint` into this function, instead of splitting out the components for the function parameters and then recreating the `JSOutPoint` here.

Reply:
Your suggestion does seem cleaner but the pattern used here mirrors `IsLockedCoin`. I'm fine with either way but I feel like maybe since these methods for notes are closely mirroring the ones for coins we should maintain this pattern? I see this specifically being used in the `AvailableCoins` method to iterate through all available coins by `vout`, not sure if there is a corollary to doing this with notes: 

https://github.com/braddmiller/zcash/blob/b3a656cf535df043450cca4dc92a9137adc773f5/src/wallet/wallet.cpp#L2263

Reply:
Eh, I'm not strongly opposed to doing it this way, and the mirroring argument is reasonable. Leave it as-is, and we can always refactor it later.